lib/src/scene_container.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+
+import 'multifinger_long_press_gesture.dart';
+
+/// Wraps [child] in a MediaQuery whose properties (such as textScale and
+/// brightness) are editable using an on-screen widget.
+///
+/// The environment editing widget can be toggled using a two-finger long press.
+class SceneContainer extends StatefulWidget {
+  final Widget child;
+
+  const SceneContainer({super.key, required this.child});
+
+  @override
+  State<SceneContainer> createState() => _SceneContainerState();
+}
+
+class _SceneContainerState extends State<SceneContainer> {
+  bool _showEnvPanel = true;
+
+  double _textScale = 1;
+  bool _isDarkMode = false;
+  TargetPlatform? _targetPlatform;
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiTouchLongPressGestureDetector(
+      numberOfTouches: 2,
+      onGestureDetected: () => setState(() {
+        _showEnvPanel = !_showEnvPanel;
+      }),
+      child: Stack(
+        children: [
+          MediaQuery(
+            data: MediaQuery.of(context).copyWith(
+              textScaleFactor: _textScale,
+              platformBrightness:
+                  _isDarkMode ? Brightness.dark : Brightness.light,
+            ),
+            child: Theme(
+              data: Theme.of(context).copyWith(platform: _targetPlatform),
+              child: widget.child,
+            ),
+          ),
+          if (_showEnvPanel)
+            SafeArea(
+              child: Align(
+                alignment: Alignment.bottomCenter,
+                child: Material(
+                  color: Colors.white,
```

Comment:
P3: should this be themed too? 🤷‍♂️

Reply:
That's a good question. I could imagine applying a dark/light mode here, but otherwise, I think it would be kind of annoying for the icons to grow with the text scale, for example – what you really want to see change is the subject of the scene.